invoice_margin/hooks.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+def _create_column(cr, table_name, column_name, column_type):
+    req = "ALTER TABLE %s ADD COLUMN %s %s" % (
+        table_name, column_name, column_type)
+    cr.execute(req)
+
+
+def pre_init_hook(cr):
+    _logger.info(
+        "Compute account_invoice_line.purchase_price for existing lines")
```

Comment:
if you want use sql to faster execution you should add a docstring to justify this method